### PR TITLE
OBSDEF-14004: Restructured files, added common variables and styles

### PR DESCRIPTION
**Pull request by ShitalSD**
OBSDEF-14004
Introduce the best practices and standards into dell-design-common repository, so that all the wrapper css follow the standards properly.
Added common variables, mixins for following items
font weight
font sizes
borders
colors
margins
paddings
Font Family should be only defined at top level, individual css overrides should not use different font families
Font Sizees, Margins, paddings should use REMs and EMs only. Dont use PX for these.
The widths and Heights of any component, pages should continue to use PX.

### CODE REVIEW

File: src/styles/common/_layout.scss [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (psivakumaran) [inline src/styles/common/_layout.scss:2]: Please change year to 2022

File: src/styles/common/_mixins.scss [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ @import "./../common/colors";
```
Comment (psivakumaran) [inline src/styles/common/_mixins.scss:11]: Please use absolute imports

**Review by psivakumaran — APPROVED** [review]
LGTM.

Please change the year from 2021 to 2022 wherever applicable
Also, if possible we can use absolute imports

**Review by GauravMenezes-Professional — APPROVED** [review]

File: src/components/button/Button.stories.tsx [hunk 3]
```diff
@@ -30,7 +30,7 @@ storiesOf("Button", module)
             </div>))
     .add("Button Sizes", () => (
         <div>
-          <Button key="primary" primary onClick={action("Small button click")}>
+          <Button key="primary" primary className="small-button-size" onClick={action("Small button click")}>
```
Comment (maheshkulkarni4) [inline src/components/button/Button.stories.tsx:33]: We should not implement any new class.

File: src/styles/common/_borders.scss [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+// Borders
+$border-default: 1px;
+$border-medium: 2px;
```
Comment (maheshkulkarni4) [inline src/styles/common/_borders.scss:14]:
```suggestion
$border-thin: 2px;
```

File: src/styles/common/_borders.scss [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+$border-thick: 4px;
+
+// Button
+$btn-border-radius: 2.5px;
+
+//Input
+$input-focus-border-radius: 0rem;
+$input-disabled-border-radius: 0rem;
```
Comment (maheshkulkarni4) [inline src/styles/common/_borders.scss:22]: Add empty lines at bottom... SASS Linters cry for this.

File: src/styles/common/_colors.scss [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+// Dell Design Standard Color codes
+
+$primary-white: #FFFFFF;
+$gray-100: #F7F7F7;
+$gray-200: #EEEEEE;
+$gray-300: #E4E4E4;
+$gray-450:  #BBBBBB;
```
Comment (maheshkulkarni4) [inline src/styles/common/_colors.scss:17]: 450, 550 ?   go with 500, 600

File: src/styles/common/_colors.scss [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+$gray-550: #888888;
+$gray-700: #666666;
+$gray-900: #333333;
+
+$blue-50: #E7F2FB;
+$blue-450: #6BACDE;
```
Comment (maheshkulkarni4) [inline src/styles/common/_colors.scss:23]: whats logic of using 45 ?

File: src/styles/common/_colors.scss [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+$primary-white: #FFFFFF;
+$gray-100: #F7F7F7;
+$gray-200: #EEEEEE;
+$gray-300: #E4E4E4;
+$gray-450:  #BBBBBB;
+$gray-550: #888888;
+$gray-700: #666666;
+$gray-900: #333333;
+
+$blue-50: #E7F2FB;
```
Comment (maheshkulkarni4) [inline src/styles/common/_colors.scss:22]:
```suggestion
$blue-400: #E7F2FB;
```

File: src/styles/common/_colors.scss [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+// Checkbox color
+$checkbox-default-color: $blue-600;
+
+// Input colors
+$input-color: $gray-550;
+$input-placeholder-color: $gray-450;
+$input-border-color: $gray-550;
+$input-error-color: $red-700;
+$input-label-color: $gray-900;
```
Comment (maheshkulkarni4) [inline src/styles/common/_colors.scss:69]: Add empty line

File: src/styles/common/_fonts.scss [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+//Primary Font
+$dell-font: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+
+// Dell Design Standards Font Weights
+$font-weight-light: 300;
```
Comment (maheshkulkarni4) [inline src/styles/common/_fonts.scss:15]: Light should be 200 ?

File: src/styles/common/_mixins.scss [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+ @import "./../common/layout";
+ @import "./../common/borders";
+ @import "./../common/fonts";
+
+
+ @mixin link_button {
```
Comment (maheshkulkarni4) [inline src/styles/common/_mixins.scss:17]: Please add documentation for each mixin, when to use each mixin

File: src/styles/components/DataGrid.scss [hunk 12]
```diff
@@ -7,40 +7,42 @@
...
+                height: $grid-column-separator-height;
+                background-color: $grid-selected-background-color;
+                margin-top: $grid-column-separator-margin-top;          
             }
         }
     }
 
     .datagrid-row:hover {
-        background-color: $dell-white;
+        background-color: $grid-backgroud-color;
```
Comment (maheshkulkarni4) [inline src/styles/components/DataGrid.scss:41]:
```suggestion
        background-color: $grid-background-color;
```

**Review by maheshkulkarni4 — APPROVED** [review]
Good Job @ShitalSD . This is really great way to start the development of common Component Repo.

File: src/components/button/Button.stories.tsx (hunk 3 above)
Comment (ShitalSD) [reply src/components/button/Button.stories.tsx:33]: Actually, observed this class is present in DDS and was missing in Button wrapper CSS so added a class in wrapper and here in story.

File: src/styles/common/_colors.scss (hunk 6 above)
Comment (ShitalSD) [reply src/styles/common/_colors.scss:17]: Followed naming convention inline with DDS, I have only included the colors which are currently used https://web.ddsproduct.com/31b3fd8b1/p/60290f-colors

File: src/styles/common/_colors.scss (hunk 7 above)
Comment (ShitalSD) [reply src/styles/common/_colors.scss:23]: DDS codes :-) https://web.ddsproduct.com/31b3fd8b1/p/60290f-colors

File: src/styles/common/_colors.scss (hunk 8 above)
Comment (ShitalSD) [reply src/styles/common/_colors.scss:22]: Same as above, there are many shades for each color so sticking with DDS convention https://web.ddsproduct.com/31b3fd8b1/p/60290f-colors

File: src/styles/common/_fonts.scss (hunk 10 above)
Comment (ShitalSD) [reply src/styles/common/_fonts.scss:15]: https://web.ddsproduct.com/31b3fd8b1/p/578bb9-typography/b/195c15
Followed this as per DDS Light 300, Regular 400, Medium 500

File: src/styles/common/_layout.scss (hunk 1 above)
Comment (ShitalSD) [reply src/styles/common/_layout.scss:2]: Done

File: src/styles/common/_borders.scss (hunk 4 above)
Comment (ShitalSD) [reply src/styles/common/_borders.scss:14]: Done

File: src/styles/common/_mixins.scss (hunk 11 above)
Comment (ShitalSD) [reply src/styles/common/_mixins.scss:17]: Done

File: src/styles/components/DataGrid.scss (hunk 12 above)
Comment (ShitalSD) [reply src/styles/components/DataGrid.scss:41]: Done

File: src/styles/common/_colors.scss (hunk 9 above)
Comment (ShitalSD) [reply src/styles/common/_colors.scss:69]: Done

File: README.md [hunk 13]
```diff
@@ -81,3 +81,32 @@ $ npm run storybook
...
+1. SCSS File Structure
+styles
+ -> Common
+ -> Components
+ index.scss
+Common utility styles such as Variables, Mixins, Functions, Fonts, Layout (padding, margin) etc. under the Common folder and reusable component styles are under Components like Buttons, Inputs etc.
+
+2. Use more generic to specific, semantically-named variables with kebab-case for example:
+Avoid- $blue-border, $light-blue-border has no pattern
+
```
Comment (akate1) [inline README.md:95]: can we remove this empty line ?

File: README.md [hunk 14]
```diff
@@ -81,3 +81,32 @@ $ npm run storybook
...
+Prefer - $border-blue, $border-light-blue 
+This pattern helps in recollection and hinting by editor.
+In this pattern you can start generically and get more specific as the text editor auto-suggests variable names.
+3. Declare/ indent the related CSS properties together so it helps to identify repeated blocks for reuse.
```
Comment (akate1) [inline README.md:99]: Please add empty line above point 3

File: README.md [hunk 15]
```diff
@@ -81,3 +81,32 @@ $ npm run storybook
...
+For example - all margin related stlyes (top, left, right etc) should be declared together, font-size, font-weight etc
```
Comment (akate1) [inline README.md:100]: stlyes -> styles

File: src/styles/common/_layout.scss (hunk 1 above)
Comment (akate1) [inline src/styles/common/_layout.scss:2]: 2021 -> 2022

File: src/styles/common/_layout.scss [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+$subtext-error-margin-left: -17.5rem;
+
+// Grid
+$grid-line-height: $line-height-20;
+$grid-cell-padding-right: 1.5rem;
+$grid-cell-padding-left: 1rem;
+$grid-column-height: 3.38rem;
+$grid-filter-top: 0.44rem;
+$grid-footer-btn-width: 1.25rem;
+$grid-container-m-top: 0.19rem;
```
Comment (akate1) [inline src/styles/common/_layout.scss:43]:
```suggestion
$grid-container-margin-top: 0.19rem;
```

File: src/styles/components/Button.scss [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+ * Copyright (c) 2021 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ @import "./../common/mixins";
```
Comment (akate1) [inline src/styles/components/Button.scss:11]: Please use absolute imports

File: src/styles/components/Input.scss [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+@import "./../common/borders";
+@import "./../common/fonts";
+
+.clr-control-container {
+    font-family: $dell-font;
+    display: inline-flex;
+    margin-bottom: $input-container-margin;
+    margin-top: $input-container-margin;
+
+    input {
```
Comment (akate1) [inline src/styles/components/Input.scss:22]: @singha158ecs  I think we need to override `clr-input` class and not input element itself.

File: src/styles/components/Input.scss [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+        }
+    }
+
+    &.clr-form-control-disabled{
+        label{
+            color: $disabled-color;
+        }
+    }
+
+    label {
```
Comment (akate1) [inline src/styles/components/Input.scss:89]: @singha158ecs Need to override class associated with input label and not HTML label itself.
@ShitalSD can we add this in guidelines ?

File: README.md (hunk 13 above)
Comment (ShitalSD) [reply README.md:95]: Done

File: README.md (hunk 14 above)
Comment (ShitalSD) [reply README.md:99]: Done

File: src/styles/common/_layout.scss (hunk 1 above)
Comment (ShitalSD) [reply src/styles/common/_layout.scss:2]: Done

File: src/styles/common/_layout.scss (hunk 16 above)
Comment (ShitalSD) [reply src/styles/common/_layout.scss:43]: Done

File: src/styles/common/_mixins.scss (hunk 2 above)
Comment (ShitalSD) [reply src/styles/common/_mixins.scss]: Tried webpack config change for this but it's not working currently with the recommended config, not sure if it's a sass-loader version issue, I will address this in separate PR.

**Pull request merged**